### PR TITLE
[nlohmann-json] Update to 3.9.0

### PR DESCRIPTION
**Pull request by thewhiteninja**
Update nlohmann-json to 3.9.0

### CODE REVIEW

File: ports/nlohmann-json/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: nlohmann-json
-Version: 3.8.0
+Version: 3.9.0
 Port-Version: 2
```
Comment (strega-nil) [inline ports/nlohmann-json/CONTROL:3]:
```suggestion
```

**Comment by thewhiteninja** [comment]
Does anyone know why the checks keep failling/hanging ?

**Comment by strega-nil** [comment]
It is way way way too easy to hit "close and comment" when you mean to hit "comment". I'm so sorry -.-

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by strega-nil** [comment]
@thewhiteninja it looks like we should update xeus as well; I think that we need to either patch it to use `/permissive-`, or `#include <ciso646>`

**Comment by thewhiteninja** [comment]
Is it ok if I add a commit to this PR to add <ciso646> to xeus ?
If yes, the patch is ready :)

**Comment by strega-nil** [comment]
@thewhiteninja yeah

**Pull request merged**